models.h

```diff
@@ -72,3 +72,17 @@ const std::map<std::wstring, std::wstring> models = {
     {L"ED-SB7",  L"gen3_7.bin"}, // SB7
     {L"ED-SR3",  L"gen3_7.bin"}, // SR3
 };
+
+typedef enum
+{
+    UnknownGen,
+    Gen1,
+    Gen2,
+    Gen3_1,
+    Gen3_2,
+    Gen3_3,
+    Gen3_4,
+    Gen3_5,
+    Gen3_6,
+    Gen3_7,
+} Brain_Gen;
```

Comment:
(must) Please use CamelCase to name enum types. While this may lead to a collision of naming convention with function names etc., it won't be a major problem since this program rarely be edited or extended.
```suggestion
} BrainGen;
```

(IMO) How about simplifying the enum as I suggested above? For example:
```c
typedef enum
{
    UnknownGen,
    Gen1,
    Gen2Or3,
} BrainGen;
```

Reply:
> Please use CamelCase to name enum types. 

Which naming convention does this project follow?

Reply:
From the review comment:
> We have no firm consensus on what naming rule to obey, but we inherit what former developers did.

Enumeration is a new thing in this repository. So let's take a look into famous naming conventions:

 - [LLVM](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly)
   - `EnumTypeName`
 - [Google](https://google.github.io/styleguide/cppguide.html#Enumerator_Names)
    - `EnumTypeName`
 - [GNU](https://www.kernel.org/doc/html/v4.10/process/coding-style.html#naming)
    - `enum_type_name`

The style `Enum_Type_Name` you introduced looks so rare that I have personally never seen it before. Since we should follow a common way to name enums, `BrainGen` is the most preferred option here.